core/commands/patch.go

```diff
@@ -0,0 +1,132 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	mh "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multihash"
+	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+	u "github.com/ipfs/go-ipfs/util"
+)
+
+var PatchCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Create a new merkledag object based on an existing one",
+		ShortDescription: `
+'ipfs patch <root> [add-link|rm-link] <args>' is a plumbing command used to
+build custom DAG objects.
+
+Examples:
+
+	EMPTY_DIR=QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn 
```

Comment:
And we shouldn't require folks to remember this magic hash.  That's what:

    $ ipfs object new unifs-dir

is for.  So:

    EMPTY_DIR=$(ipfs object new unixfs-dir)